pkg/images/enricher/enricher_impl.go

```diff
@@ -164,7 +164,8 @@ func (e *enricherImpl) delegateEnrichImage(ctx context.Context, enrichCtx Enrich
 	}
 
 	// Send image to secured cluster for enrichment.
-	scannedImage, err := e.scanDelegator.DelegateScanImage(ctx, image.GetName(), clusterID, enrichCtx.FetchOpt.forceRefetchCachedValues())
+	force := enrichCtx.FetchOpt.forceRefetchCachedValues() || enrichCtx.FetchOpt == UseImageNamesRefetchCachedValues
```

Comment:
There is also a fetch option called `IgnoreExistingImages`. Should that option be considered when deciding whether to force refetch?
I am not very much aware of the enrichment workflow, so this may not be relevant here. Please ignore if irrelevant.

Reply:
That logic was meant to use whatever fetchopts are set when ad-hoc (roxctl, jenkins, API, etc.) requests receive the `force` flag. (see [here](https://github.com/stackrox/stackrox/blob/f693e5b1360d608f91c531f425b69538a732586f/central/image/service/service_impl.go#L303-L305) for example)

If we change this logic to treat `IgnoreExistingImages` the same as force then the cache'd results would never be used since `IgnoreExistingImages` is set by default on the various ad-hoc requests (which was desired for delegated scanning).

We have ROX-17548 to clean them up in the future.

Reply:
That makes sense, thanks.